### ISSUE TITLE
Psalm does not catch making a non-abstract method abstract in child class

### ISSUE THREAD

**Issue by nickyr** [vimeo/psalm#566]
Paste the following into getpsalm.org. It thinks everything is fine (except it thinks all the classes aren't used--maybe another bug?)

```php
<?php
class BaseClass {
    public function foo(): int {
        return 3; 
    }
}

abstract class Child extends BaseClass {
    abstract public function foo(): int;
}

class GrandChild extends Child {
    public function foo(): int {
        return 4; 
    }
}

$c = new GrandChild();
echo $c->foo();
```

However, if you tried to actually run this, PHP would throw a fatal error:

```
PHP Fatal error:  Cannot make non abstract method BaseClass::foo() abstract in class Child
```

Psalm should catch this.
